### ISSUE TITLE
Infra Container Enhancement

### ISSUE THREAD

**Issue by cdoern** [containers/podman#10933]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

`InfraContainerConfig` is not as parallel as it should be to `ContainerConfig`. There should be clear differences as certain attributes do not apply. However, for simplicity and readability I believe we should begin a slight rework of what `InfraContainerConfig` looks like. 

Ran into this while working on `pod create` flags. Certain things that can be done while creating a regular container cannot be done while creating infra due to a lot of missing attributes. This is a good foundational step before building up `podman pod create`. A good specific example is while implementing the `--volume` flag, I could not use overlayVolumes because `overlay.Mount()` requires the usage of the containers `StaticDir` which is where persistent libpod data is stored. Infra does not have this. This change led me down a rabbit hole of other changes that could be made which leads me to think a general rework is called for.

Will link commits I make to my infraEnhance branch so others can see what I am suggesting.

Open for suggestions and discussions!

**Comment by mheon** [comment]
The StaticDir isn't something we know before the container is created - it's configured by c/storage as part of container creation?

**Comment by cdoern** [comment]
hmmm okay @mheon I guess the issue is there is not equiv place as there is w/ regular containers to do specific mount such as overlay which uses the `StaticDir`

**Comment by rhatdan** [comment]
Doesn't the infra container get a staticdir?

**Comment by cdoern** [comment]
Yes @rhatdan it does but not it cannot be explicitly set until infra goes through the normal creation process. I am brainstorming on how to consolidate infra and regular container creation. I feel like they should be more comparable and easy to track. Gonna look into this more later today and tomorrow.
